Issue converting BGR image to RGB

##### System information (version)
OpenCV => 3.1
Operating System / Platform => MacOS 10.13.6
Compiler => python3 via terminal command


##### Detailed description

Hello, I am experiencing a very weird error when trying to convert BGR images written to my disk with:
cv2.imwrite(image_path, frame)

The frame variable is a raw capture from cv2.VideoCapture(0) that I write to disk. The actual error occurs when reading in this image. Here is some example code:
image = cv2.imread(imagePath)
rgb = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)

Here is my error:
Traceback (most recent call last):
  File "face_trainer.py", line 30, in <module>
    rgb = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
cv2.error: OpenCV(3.4.2) /Users/travis/build/skvark/opencv-python/opencv/modules/imgproc/src/color.hpp:253: error: (-215:Assertion failed) VScn::contains(scn) && VDcn::contains(dcn) && VDepth::contains(depth) in function 'CvtHelper'

Can someone please try to help me interpret this error?

##### Steps to reproduce
``` 
from imutils import paths
import face_recognition
import argparse
import pickle
import cv2
import os

default_encoding_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "encodings.pickle")

ap = argparse.ArgumentParser()
ap.add_argument("-i", "--dataset", required=True, help="path to input directory of faces + images")
ap.add_argument("-e", "--encodings", default=default_encoding_path,
                help="path to serialized db of facial encodings")
ap.add_argument("-d", "--detection-method", type=str, default="hog",
                help="face detection model to use: either `hog` or `cnn`")
args = vars(ap.parse_args())

print("[INFO] quantifying faces...")
imagePaths = list(paths.list_images(args["dataset"]))
knownEncodings = []  # initialize the list of known encodings and known names
knownNames = []

# loop over the image paths
for (i, imagePath) in enumerate(imagePaths):
    print(imagePath)
    # extract the person name from the image path
    print("[INFO] processing image {}/{}".format(i + 1, len(imagePaths)))
    name = imagePath.split(os.path.sep)[-2]
    image = cv2.imread(imagePath)
    rgb = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
    boxes = face_recognition.face_locations(rgb, model=args["detection_method"])
    encodings = face_recognition.face_encodings(rgb, boxes)
    for encoding in encodings:
        knownEncodings.append(encoding)
        knownNames.append(name)

print("[INFO] serializing encodings...")  # writing the facial encodings and names to the disk
data = {"encodings": knownEncodings, "names": knownNames}
f = open(args["encodings"], "wb")
f.write(pickle.dumps(data))
f.close()
```

The image you are trying to convert is probably not a three-channel image. My guess is that either it is not loaded at all, or it was saved and loaded as a single-channel (grayscale) image:

```
>>> img = cv2.imread('/dev/null')
>>> print(img)
None
>>> img2 = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cv2.error: OpenCV(3.4.2-dev) /home/rok/Projects/opencv/modules/imgproc/src/color.hpp:253: error: (-215:Assertion failed) VScn::contains(scn) && VDcn::contains(dcn) && VDepth::contains(depth) in function 'CvtHelper'
```
```
>>> img = cv2.imread('three_channel_image.jpg')
>>> img_g = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
>>> img2 = cv2.cvtColor(img_g, cv2.COLOR_BGR2RGB)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cv2.error: OpenCV(3.4.2-dev) /home/rok/Projects/opencv/modules/imgproc/src/color.hpp:253: error: (-215:Assertion failed) VScn::contains(scn) && VDcn::contains(dcn) && VDepth::contains(depth) in function 'CvtHelper
```

For what is worth, the message in previous versions of OpenCV was a bit more direct:
```
>>> img = cv2.imread('/dev/null')
>>> img2 = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
OpenCV(3.4.1) Error: Assertion failed (scn == 3 || scn == 4) in cvtColor, file /builddir/build/BUILD/opencv-3.4.1/modules/imgproc/src/color.cpp, line 11115
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cv2.error: OpenCV(3.4.1) /builddir/build/BUILD/opencv-3.4.1/modules/imgproc/src/color.cpp:11115: error: (-215) scn == 3 || scn == 4 in function cvtColor
```

--- a/modules/imgproc/src/color.cpp
+++ b/modules/imgproc/src/color.cpp
@@ -178,6 +178,8 @@ void cvtColor( InputArray _src, OutputArray _dst, int code, int dcn )
 {
     CV_INSTRUMENT_REGION()
 
+    CV_Assert(!_src.empty());
+
     if(dcn <= 0)
             dcn = dstChannels(code);
 

--- a/modules/imgproc/src/color.hpp
+++ b/modules/imgproc/src/color.hpp
@@ -247,10 +247,14 @@ struct CvtHelper
 {
     CvtHelper(InputArray _src, OutputArray _dst, int dcn)
     {
+        CV_Assert(!_src.empty());
+
         int stype = _src.type();
         scn = CV_MAT_CN(stype), depth = CV_MAT_DEPTH(stype);
 
-        CV_Assert( VScn::contains(scn) && VDcn::contains(dcn) && VDepth::contains(depth) );
+        CV_Check(scn, VScn::contains(scn), "Invalid number of channels in input image");
+        CV_Check(dcn, VDcn::contains(dcn), "Invalid number of channels in output image");
+        CV_CheckDepth(depth, VDepth::contains(depth), "Unsupported depth of input image");
 
         if (_src.getObj() == _dst.getObj()) // inplace processing (#6653)
             _src.copyTo(src);

